src/backend/distributed/commands/multi_copy.c

```diff
@@ -986,6 +1023,50 @@ OpenCopyTransactions(CopyStmt *copyStatement, ShardConnections *shardConnections
 
 
 /*
+ * BinaryCopyFormat iterates over columns of the relation given in rowOutputState and
+ * looks for a column whose type is array of user-defined type. If it founds that
+ * means we cannot use binary format for COPY and it return false.
+ */
+static bool
+BinaryCopyFormat(TupleDesc tupleDescription, CopyOutState rowOutputState)
+{
+	bool binaryCopyFormat = true;
+	uint32 totalColumnCount = (uint32) tupleDescription->natts;
+	uint32 columnIndex = 0;
+
+	MemoryContext oldContext = MemoryContextSwitchTo(rowOutputState->rowcontext);
```

Comment:
Just talked with Andres about this, I think this is unnecessary complication. It's just a few Oid's which are going to be released at the end of the transaction anyway, we don't really care if they're held onto a little longer than necessary.